lab5.py

```diff
@@ -0,0 +1,93 @@
+"""This module contains classes for working with geometric figures in 2D space"""
+import math
+from enum import Enum
+
+class Colour(Enum):
+    """Class representing different colors"""
+    ONE = "RED"
+    TWO = "GREEN"
+    THREE = "BLUE"
+
+class Point:
+    """Class representing a point in 2D space"""
+    def __init__(self, x, y):
+        """Initialize a point with given x and y coordinates"""
+        self.x = x
+        self.y = y
+
+    def get_x(self):
+        """Get the x-coordinate"""
+        return self.x
+
+    def get_y(self):
+        """Get the y-coordinate"""
+        return self.y
+
+class Polynom:
+    """Class representing a polygon with methods to calculate perimeter/longest diagonal"""
+    def __init__(self, colour):
+        """Initialize a polygon with a specified color"""
+        self.points = []
+        self.colour = colour
+
+    def add_point(self, point):
+        """Add a point to the polygon"""
+        self.points.append(point)
+
+    def perimeter(self):
+        """Calculate the perimeter of the polygon"""
+        perimeter = 0
+        for i, _ in enumerate(self.points):
+            x = self.points[(i+1) % len(self.points)].get_x() - self.points[i].get_x()
+            y = self.points[(i+1) % len(self.points)].get_y() - self.points[i].get_y()
+            perimeter += math.sqrt(x**2 + y**2)
+        return perimeter
+
+    def longest_diagonal(self):
+        """Find and calculate the longest diagonal in the polygon"""
+        longest_diagonal = 0
+        values = []
+        for i, _ in enumerate(self.points):
+            for j in range(i + 2,len(self.points)):
+                x = self.points[i].get_x() - self.points[j].get_x()
+                y = self.points[i].get_y() - self.points[j].get_y()
+                diagonal = math.sqrt(x**2 + y**2)
+                values.append(diagonal)
+                longest_diagonal = max(values)
+        return longest_diagonal

```

Comment:
calculate longest diagonal once, if you've decided to store all diagonals into array 

```suggestion
        return max(values)

```

Reply:
Okay, I'll fix that. Thx for review.